### ISSUE TITLE
Support for Weights & Biases (WandB) ?

### ISSUE THREAD

**Issue by awsaf49** [ultralytics/ultralytics#206]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

It appears that the YOLOv5 project has recently discontinued support for Weights & Biases (WandB) and has chosen to support ClearML for its experimentation tracking and visualization needs. It is unclear whether there will be any future collaboration between YOLOv8 and WandB.  Also, will it be allowed to create pull requests before collaboration? I would love to see YOLOv8 with WandB tracking.

**Comment by geezacoleman** [comment]
I have been able to use WandB tracking with training YOLOv8 just by following the same procedure as for v5:

```
# Weights & Biases  (optional)
%pip install -q wandb
import wandb
wandb.login()
```

**Comment by awsaf49** [comment]
@geezacoleman  that is weird as I couldn't find **wandb** in the code base. Also in the callback there is **comet and clearml** but not **wandb**,

![image](https://user-images.githubusercontent.com/36858976/211711595-6aed3dab-cd2d-46d3-9038-518c2abad4e1.png)

**Comment by geezacoleman** [comment]
That's what I found strange too - but you get what seem like reasonable results that check out with the validation results provided once it's finished training. Some features (val/loss) don't work but the rest seems to be ok. 

I must be missing something with either how wandb works or an import somewhere - can't be magic!

**Comment by AyushExel** [comment]
Yes guys wandb is not officially supported. The initial code there was for running tests

**Comment by awsaf49** [comment]
@AyushExel  any chance of future support of WandB?

**Comment by geezacoleman** [comment]
I must have cloned it before the cleanup yesterday. It seems you can find the relevant file here, if you wanted to add it back in: https://github.com/ultralytics/ultralytics/commit/c8e3c5db4bc737cd9ed8e63ae71dc4949cb6553f#diff-1365dac64b857511656e6f9948f7c04022f4a9c14f7732ee26b4ca29d930f0a2

**Comment by awsaf49** [comment]
Now it make sense @geezacoleman

**Comment by AyushExel** [comment]
> @AyushExel any chance of future support of WandB?

Yes of course. Anything is possible in the future. stay tuned! :)

**Comment by sstainba** [comment]
> > @AyushExel any chance of future support of WandB?
> 
> Yes of course. Anything is possible in the future. stay tuned! :)

I'd like to second this... Please add support for this back in.
